content/blog/pulumi-azure-2-0/_index.md

```diff
@@ -0,0 +1,149 @@
+
+## New Resources for Virtual Machine & Virtual Machine Scale Set
+
+There are four new resources for Virtual Machine & Virtual Machine Scale Set separated by operating system: `LinuxVirtualMachine`, `WindowsVirtualMachine`, `LinuxVirtualMachineScaleSet`, `WindowsVirtualMachineScaleSet`. The old-style generic `VirtualMachine` and `VirtualMachineScaleSet` are still available, so you can take your time before migrating existing stacks.
+
+Here is a snippet that defines an Ubuntu VM:
+
+``` ts
+const vm = new azure.compute.LinuxVirtualMachine("server-vm", {
+    resourceGroupName,
+    networkInterfaceIds: [networkInterface.id],
+    size: "Standard_A0",
+    sourceImageReference: {
+        publisher: "Canonical",
+        offer: "UbuntuServer",
+        sku: "16.04-LTS",
+        version: "latest",
+    },
+    osDisk: {
+        caching: "ReadWrite",
+        storageAccountType: "Standard_LRS",
+    },
+    computerName: "hostname",
+    adminUsername: username,
+    adminPassword: password,
+    disablePasswordAuthentication: false,
+);
+```
+
+## Removing Deprecated Resources, Invokes, and Fields
+
+Azure Active Directory has its [own Pulumi provider](https://github.com/pulumi/pulumi-azuread/), so all the AD resources were now removed from the `AD` namespace of the Azure provider 2.0.
+
+A bunch of other resources, invokes, and fields were removed too, following the changes in the upstream provider. You can see the full list in [this upgrade guide](https://www.terraform.io/docs/providers/azurerm/guides/2.0-upgrade-guide.html).
+
+## Default Versions in Callback Functions
+
+[Serverless as Callbacks](https://www.pulumi.com/blog/serverless-as-simple-callbacks-with-pulumi-and-azure-functions/) running on Azure Functions have bumped the default version of Azure Functions runtime to `~3` and `Node.js` to `~12`. As always, you can override the default to set the versions that you require.
+
+Callback Functions also moved from `ZipBlob` to `Blob` deploy as described above, so expect another replacement operation while upgrading.
+
+## Custom Timeouts
+
+Some types of resources take longer to create than others. For instance, provisioning a new Cosmos DB account may take from 10 minutes to over an hour depending on the number of geo locations and other factors.
+
+If you find that creation of a target resource times out, you can override the default timeout using resource options:
+
+``` ts
+const cosmosdbAccount = new azure.cosmosdb.Account("cosmosdb", {
+    resourceGroupName: resourceGroup.name,
+    offerType: "Standard",
+    geoLocations: manyLocations,
+    consistencyPolicy: { consistencyLevel: "Session" },
+}, {
+    customTimeouts: {
+        create: "30m",
+        update: "60m",
+    },
+});
+```
+
+You shouldn't need this ability often, but it can be a lifesaver when you do.
+
+## Getting Started with Migration
+
+To get started with Azure provider 2.0, update the versions in your package manager:
```

Comment:
Would it make sense to use the language chooser here?

Reply:
Great idea!